File: leetcode-js/370-range-addition.js
/**
 * @param {number} length
 * @param {number[][]} updates
 * @return {number[]}
 */
var getModifiedArray = function(length, updates) {
    const result = Array(length).fill(0);

    for (const [start, end, value] of updates) {
        result[start] += value;

        if (end < length - 1) {
            result[end + 1] -= value;
        }
    }

    let sum = 0;

    for (let i = 0; i < length; i += 1) {
        sum += result[i];
        result[i] = sum;
    }

    return result;
};

File: leetcode-java/370-range-addition.java
public class Solution {
    public int[] getModifiedArray(int length, int[][] updates) {
        int[] res = new int[length];

        for (int[] update : updates) {
            int start = update[0];
            int end = update[1];
            int inc = update[2];

            res[start] += inc;

            if (end < length - 1) {
                res[end + 1] -= inc;
            }
        }

        int sum = 0;

        for (int i = 0; i < res.length; i++) {
            sum += res[i];
            res[i] = sum;
        }

        return res;
    }
}
